Multiple dirs class set

hgraca opened the pull request:
Currently, if we want to include two folders from the project root, we need to add the root of the project and then exclude everything we dont want to include, and remember to add a folder to the config if we add a new folder to the project root.

With the `MultipleDirClassSet` we can specify just the directories we want it to scan,
removing all cruft from our config.

jdreesen commented:
How about refactoring `ClassSet` so that it can handle multiple directories instead? 

The constructor is `private` so we could change it (without BC break) and add a second static constructor `fromDirs(string ...$directoryList)`. It shouldn't be a breaking change, right?

hgraca commented:
Yes! That was my first approach, but I found some reason to do it differently... Dont remember what it was though. :(

We can even change the named constructor without a BC break.

my only concern here is with the `getDir()` method, cozz its return will not be atomic anymore, but on the other hand we are only using it for writing it on the screen.

Anyway, as I can't see anymore what drove me to this approach, I will simplify as you suggest.

hgraca commented:
@jdreesen done, and force pushed

AlessandroMinoccheri commented on src/ClassSet.php:
```diff
@@ -7,15 +7,15 @@
-    /** @var string */
-    private $directory;
+    /** @var string[] */
+    private $directoryList;
```
I was thinking of a var name like `directories` instead of `directoryList`, what do you think?

hgraca replied:
It does read better, however when plural prefer to suffix with "List", otherwise in most cases the difference between plural and singular will be only a letter (s), wich can easily be overlooked and result in confusion.

That said, if my argument doesn't convince you let me know and I will gladly change it. :)

And tkx for the CR btw.

AlessandroMinoccheri commented:
@hgraca Before approving, can you please add something to the readme file :) ?

hgraca commented:
@AlessandroMinoccheri done

hgraca commented:
@micheleorselli @AlessandroMinoccheri 

I rebased, fixed the conflicts, and added 2 commits.

Let me know if this is ok or if I can do something else.

micheleorselli commented:
Looks good to me, thank you @hgraca! 🙇
